gapic/src/main/com/google/gapid/perfetto/views/TraceComposite.java

```diff
@@ -82,9 +84,18 @@ public TraceComposite(Composite parent, Analytics analytics, Theme theme) {
       }
     });
     canvas.addListener(SWT.MouseHorizontalWheel, e -> {
-      if ((e.stateMask & SWT.MODIFIER_MASK) == SWT.MOD1) {
-        // Ignore horizontal scroll, only when zooming.
+      int mods = e.stateMask & SWT.MODIFIER_MASK;
+      if (mods == SWT.MOD1) {
+        // Ignore horizontal scroll, when zooming.
+        e.doit = false;
+      } else {
+        // Treat horizontal touch pad/scroll wheel ourselves, rather than going through the
+        // scrollbar, since the scrollbar's size is limited.
         e.doit = false;
```

Comment:
It looks like there is `e.doit = false` in both branches, could this be refactored as:

```
e.doit = false;
if (mods != SWT.MOD1) {
    if (state.dragX(...)) {
        ...
    }
}
```

Reply:
Done. I saw that, but had decided to not write it that way as I thought it was confusing. I think I found an ok way to write it now that explains properly why things are the way they are.